client/add/add.js

```diff
@@ -0,0 +1,52 @@
+// Controller for the add site form
+
+angular.module('divestop.addsite', [])
+  .controller('AddSiteController', function(SharedProperties, DiveSites) {
+    this.site = {};
+    this.site.features = [];
+    this.site.aquaticLife = [];
+    this.site.coordinates = SharedProperties.newSite;
+    this.showForm = SharedProperties.showForm;
+
+    this.templateUrl = 'add/add.html';
+
+    this.addFeature = function() {
+      this.site.features.push(this.newFeature);
+      this.newFeature = '';
+    };
+    this.removeFeature = function(index) {
+      this.site.features.splice(index, 1);
+    };
+    this.addAquaticLife = function() {
+      this.site.aquaticLife.push(this.newAquaticLife);
+      this.newAquaticLife = '';
+    };
+    this.removeAquaticLife = function(index) {
+      this.site.aquaticLife.splice(index, 1);
+    };
+
+    this.clearForm = function() {
```

Comment:
http://stackoverflow.com/questions/21571370/resetting-form-after-submit-in-angularjs